### DISCUSSION TITLE
Attach an email to a person

### DISCUSSION THREAD

**Discussion by Bonapara** [twentyhq/twenty#1614] [Ideas]
# Attach an email to a person

It should be possible to attach an email to a **person** from an email thread accessed from a **company** record when the email is not recognized as associated with any **Person** record in our database. (Almost nobody will manually add emails to **People** records)

Everyone that has the right to see an email, should be able to attach an email recipient to a **Person** record: 

1. A chip should be displayed over the name of a recipient that is not part of your team (we don't want to have our teammates in our CRM) and that is not already associated with a **People** record. 
2. Click on it, and a menu opens, with a search field already filled with the recipient name.
3. Clicking on the **Person** record on the menu, associate the email recipient to the given **person** record. (It means that it fills the **person** email field, and that the email will be now visible on the given **person** detail page.)

https://github.com/twentyhq/twenty/assets/19412894/65804b90-6854-4fea-b59b-75d2ad9f35c2

# Linked people

Once a record has been attached (either manually as described above, or automatically if the **Person** record has already been filled with an email, a link should appear to navigate to the **Person** detail page.

In the following example, **Alexandre Prot** has already been associated with an email on Twenty, clicking on his name, opens a chip to navigate to his detail page. 

The condition to display this chip is thus "Does a person record has his email filled in his profile? 

https://github.com/twentyhq/twenty/assets/19412894/3519d580-ec79-4e5d-8fe8-bc7dfced993e

# Condition on the email field

We should add validation to prevent users from letting two people have the same emails. (Restore the previous state with an alert message if the email is already set for another record)

**Comment by abhip167** [comment]
Can I take this ?

**Comment by charlesBochet** [comment]
@abhip167 we haven't implemented the email thread yet. I'm putting this one on-hold as we are waiting for the new backend structure to be ready (should take ~2 weeks)
